react/week2/Task/src/Todo.js

```diff
@@ -0,0 +1,73 @@
+import React, { useState } from 'react';
+import './Todo.css';
+import Task from './Task'
+
+
+
+ const Todo = () => {
+    
+    const [tasks, setTasks] = useState([
+        {
+            description: "Get out of bed",
+            completed: false
+        },
+        {
+            description: " brush teeth",
+            completed: false
+        },
+        {
+            description: "eat breakfast",
+            completed: false
+        }
+    ]);
+
+   
+
+
+    const addTask = () => {
+        const getRandomTask = tasks[Math.floor(Math.random() * tasks.length)];
+        const newTasks = [...tasks, { description:getRandomTask, completed: false }];
+        setTasks(newTasks);
+    };
+
+    const handleClick = () => {
+         addTask();
+  };
+
+    const completeTask = index => {
+        const newTasks = [...tasks];
+        newTasks[index].completed = true;
+        setTasks(newTasks);
```

Comment:
You can now mark a task as complete, but you can't afterwards mark it as incomplete (i.e. toggle it).
How about something like
```
  const toggleComplete = (index) => {
    setTasks((prev) =>
      prev.map((task, taskIndex) =>
        taskIndex === index ? { ...task, completed: !task.completed } : task
      )
    );
  };
```